apps/angular/ngfor-enhancement/src/app/ng-for-empty.directive.ts

```diff
@@ -0,0 +1,33 @@
+import {
+  Directive,
+  DoCheck,
+  Input,
+  TemplateRef,
+  ViewContainerRef,
+} from '@angular/core';
+import { EmptyListDirective } from './empty-list.directive';
+import { NgFor } from '@angular/common';
+
+@Directive({
+  // eslint-disable-next-line @angular-eslint/directive-selector
+  selector: '[ngFor]',
+  standalone: true,
+  hostDirectives: [
+    {
+      directive: NgFor,
+      inputs: ['ngForOf'],
+    },
+  ],
+})
+export class NgForEmptyDirective<T> implements DoCheck {
+  @Input() ngForOf: T[] = [];
+  @Input() ngForEmpty!: TemplateRef<EmptyListDirective>;
+
+  constructor(private vcr: ViewContainerRef) {}
+
+  ngDoCheck(): void {
+    if (!this.ngForOf.length) {
+      this.vcr.createEmbeddedView(this.ngForEmpty);
+    }
```

Comment:
I didn't test it, but I'm not sure it will work if your list is dynamic. Meaning, if you have element on delete them, I don't think you will show the empty template

Reply:
I added a delete button with a delete method that filtered the this.persons array to exclude a passed in id (added an id to Person object) and the empty prompt showed.

Reply:
Ok thanks for testing it.